torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -547,16 +547,30 @@ bool is_float_or_complex_list(PyObject* obj) {
   return true;
 }
 
-static bool is_int_list_(PyObject* obj, int broadcast_size) {
+static bool all_ints_in_tuple(PyObject* obj) {
+    for (auto i: c10::irange(PySequence_Size(obj))) {
+        auto item = py::reinterpret_steal<py::object>(
+        PySequence_GetItem(obj, i));
+      if (!THPUtils_checkIndex(item.ptr())) {
+        return false;
+      }
+    }
+
+    return true;
+}
+
+static bool is_int_list(PyObject* obj, int broadcast_size) {
   if (PyTuple_Check(obj) || PyList_Check(obj)) {
     if (PySequence_Size(obj) == 0) {
       return true;
     }
-    auto item = py::reinterpret_steal<py::object>(
-        PySequence_GetItem(obj, 0));
-    if (THPUtils_checkIndex(item.ptr())) {
+    
+    if (all_ints_in_tuple(obj)) {
```

Comment:
This is a pessimization for regular eager mode; previously only the first element of the tuple is type tested, but now we have to iterate over the entire tuple no matter what. Doing this more efficiently will probably involve an involved refactor of the code here so we may have to live with it for now, but the inefficiency here should be prominently marked.

Reply:
I'm a little perplexed why this is necessary though; based on your description of the problem (int overload was accidentally being selected preferentially over symint overload), the short circuit here doesn't feel like it would help, because if NOT all the things in the tuple are int, we still are going to do the conventional test below (which would pass you if you have (0, symint))